common/djangoapps/dark_lang/views.py

```diff
@@ -0,0 +1,153 @@
+"""
+Views file for the Darklang Django App
+"""
+from openedx.core.lib.api.view_utils import view_auth_classes
+from django.views.generic.base import View
+from django.contrib.auth.decorators import login_required
+from django.utils.decorators import method_decorator
+
+from edxmako.shortcuts import render_to_response
+
+from openedx.core.djangoapps.user_api.preferences.api import (
+    delete_user_preference, get_user_preference, set_user_preference
+)
+from lang_pref import LANGUAGE_KEY
+
+from django.utils.translation import LANGUAGE_SESSION_KEY
+from django.utils.translation import ugettext as _
+
+from dark_lang import DARK_LANGUAGE_KEY
+from dark_lang.models import DarkLangConfig
+
+LANGUAGE_INPUT_FIELD = 'preview_lang'
+
+
+@view_auth_classes()
+class DarkLangView(View):
+    """
+    View used when a user is attempting to change the preview language using Darklang.
+
+    Expected Behavior:
+    GET - returns a form for setting/resetting the user's dark language
+    POST - updates or clears the setting to the given dark language
+
+    Note: This class and View are meant to replace the ability to set the DarkLang in the middleware, which was
+    determined to be a security risk. (TNL-4742)
+    """
+    template_name = 'darklang/preview_lang.html'
+
+    @method_decorator(login_required)
+    def get(self, request):
+        """
+        Displays the Form for setting/resetting a User's dark language setting
+
+        Arguments:
+            request (Request): The Django Request Object
+
+        Returns:
+            HttpResponse: View containing the form for setting the preview lang
+        """
+        context = {
+            'disable_courseware_js': True,
+            'uses_pattern_library': True
+        }
+        return render_to_response(self.template_name, context)
+
+    @method_decorator(login_required)
+    def post(self, request):
+        """
+        Sets or clears the DarkLang depending on the incoming post data.
+
+        Arguments:
+            request (Request): The Django Request Object
+
+        Returns:
+            HttpResponse: View containing the form for setting the preview lang
+        """
+        context = {
+            'disable_courseware_js': True,
+            'uses_pattern_library': True
+        }
+        (success, result) = self.process_darklang_request(request)
```

Comment:
Nit: returning a tuple with information for success is generally harder to read and maintain.  Typically, we raise exceptions on failures and catch expected ones if they need to be translated to an error.  I don't think it's a merge-stopper, but something for you to consider.